app/shared/src/commonMain/kotlin/ui/subject/episode/video/PlayerSkipOpEdState.kt

```diff
@@ -0,0 +1,83 @@
+package me.him188.ani.app.ui.subject.episode.video
+
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import me.him188.ani.app.videoplayer.ui.state.Chapter
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+@Stable
+class PlayerSkipOpEdState(
+    chapters: State<List<Chapter>>,
+    private val onSkip: (targetMillis: Long) -> Unit,
+    videoLength: State<Duration>,
+) {
+    private var currentChapter: CurrentChapter? by mutableStateOf(null) 
+    private val opEdChapters by derivedStateOf {
+        chapters.value.filter {
+            OpEdLength.fromVideoLengthOrNull(videoLength.value)
+                ?.isOpEdChapter(it.durationMillis.milliseconds) == true
+        }.map { CurrentChapter(chapter = it, true) }
+    }
+
+    val skipped: Boolean by derivedStateOf {
+        currentChapter?.skipped ?: false
+    }
+    val showSkipTips: Boolean by derivedStateOf {
+        currentChapter != null && skipped
+    }
+
+    fun cancelSkipOpEd() {
+        currentChapter = opEdChapters.find { it == currentChapter }?.apply { this.skipped = false }?.copy()
+    }
+
+
+    /**
+     * 每秒调用一次update
+     * 根据[currentPos]感知[currentPos]到5秒后这个区间是否会有章节开头，
+     * 根据当前秒的位置显示/隐藏tips，
+     * 并且如果[currentPos]在章节开头的位置，根据[skipped]跳过该章节
+     */
+    fun update(currentPos: Long) {
+        if (opEdChapters.isEmpty()) return
+        // 在显示跳过提示范围
+        opEdChapters.find { it.chapter.offsetMillis in currentPos - 1000..currentPos + 5000 }?.let {
+            if (currentChapter == null) {
+                currentChapter = it.copy()
+            }
+        } ?: run {
+            currentChapter = null
+        }
+        // 在跳过 OP/ED 范围
+        opEdChapters.map { it.chapter }.find { it.offsetMillis in currentPos - 1000..currentPos }?.run {
```

Comment:
```suggestion
        opEdChapters.asSequence().map { it.chapter }.find { it.offsetMillis in currentPos - 1000..currentPos }?.run {
```

list 的 map 会构造新的 list, 会有一些内存影响, 用 sequence 就没问题